### ISSUE TITLE
Use Optim.jl trust region code

### ISSUE THREAD

**Issue by rgiordan** [jeff-regier/Celeste.jl#237]
Our trust region code has been merged into Optim.jl, and we should use that going forward.

@jeff-regier , do you want to wait for a tagged version with the TR code or just go ahead and use the latest release?

**Comment by jeff-regier** [comment]
That's good news. Up to you, Ryan, whether to wait for the tagged release

**Comment by rgiordan** [comment]
We need to move to ForwardDiff 2.0 to do this, but we're held up by the problems described in https://github.com/jeff-regier/Celeste.jl/pull/248.

**Comment by jeff-regier** [comment]
Closed in #269
